---
title: "Differential Expression with COI"
author: "Emma Berdan"
date: "`r Sys.Date()`"
output:
   html_document:
      code_folding: hide
      df_print: paged
      highlights: pygments
      number_sections: true
      self_contained: true
      theme: default
      toc: true
      toc_float:
         collapsed: true
         smooth_scroll: true
---

# Overview

- Principal Investigator: Aditi Hazra
- Experiment: Breast Cancer patient samples

```{r, cache = FALSE, message = FALSE, warning=FALSE}
library(tidyverse)
library(knitr)
library(DESeq2)
library(DEGreport)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
library(biomaRt)
library(topGO)
library(KEGGREST)
library(Category)
library(ape)
library(data.table)
library(fgsea)
library(clusterProfiler)
library(sva)
library(gprofiler2)
library(janitor)
library(ggvenn)
library(ggplot2)
ggplot2::theme_set(theme_light(base_size = 14))
opts_chunk[["set"]](
    cache = FALSE,
    dev = c("png", "pdf"),
    error = TRUE,
    highlight = TRUE,
    message = FALSE,
    prompt = FALSE,
    tidy = FALSE,
    warning = FALSE,
    fig.height = 4)

sanitize_datatable = function(df, ...) {
 # remove dashes which cause wrapping
 DT::datatable(df, ..., rownames=gsub("-", "_", rownames(df)),
                   colnames=gsub("-", "_", colnames(df)))
}
```

# Load and pre-process data

We load our data. We look at how many zeros we have per gene and see that for many genes it is very high. We set a cutoff that at most 25 samples can have zero counts.


## Get annotations
```{r get gene_annotations}
 # library(biomaRt)
#

# Get gene annotations based on reference data version used for alignment/quantification from BiomaRt

# Specify the Ensembl release ## dataset used by bcbio (check archives if needed)
#
# # List current versions of datasets
#  ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL")
#  datasets <- listDatasets(ensembl)
# #
# # # Identify the proper archive to use for corresponding Ensembl release
#  archives <- listEnsemblArchives()
# #
# # # # Use ensembl GRCm39
#   ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL",
#                     dataset = "mmusculus_gene_ensembl")
# # #
# # # ## Build biomaRt query
#  filters = listFilters(ensembl)
#  attributes = listAttributes(ensembl)
# # #
#   gene_annotations <- getBM(attributes=c('ensembl_gene_id', 'entrezgene_id',  'external_gene_name', 'description'),
#         filters = 'ensembl_gene_id',
#        values = rownames(counts),
#         mart = ensembl)
# #
# # #remove duplicated ensembl ids with different entrez gene IDs.
#  pruned_annotations = gene_annotations %>%
#    arrange(ensembl_gene_id, ensembl_gene_id) %>%
#    distinct(ensembl_gene_id, .keep_all = TRUE)
# #
#  write.csv(pruned_annotations, "/Users/emb016/Documents/HBC04572_Rosen/data/biomart.csv")
#
 pruned_annotations <- read.csv("/Users/emb016/Documents/HBC04676_Sicinski/annotation/biomart.csv") %>% .[,-1]
```


## Load Data

```{r  show-metadata}
se <- readRDS("/Users/emb016/Documents/HBC04292_Hazra/full_data/bcbio-se.rds")

metadata <- colData(se) %>% 
    as_tibble(rownames = NULL)

metrics <- metadata(se)$metrics %>% 
    left_join(metadata, by = c("sample" = "sample"))

meta_sm <- metadata %>%
  as.data.frame() %>%
  column_to_rownames("sample")


raw_counts <- assays(se)[["raw"]] 



coding_genes <- read.table("/Users/emb016/Documents/HBC04292_Hazra/lnc_and_protein_hg38.txt", header=FALSE)

raw_counts2 <- subset(raw_counts, rownames(raw_counts) %in% coding_genes$V2) 



## Remove WU104 and WU228

raw_counts3 <- raw_counts2[,-c(3 , 96)] %>% as_tibble()  %>% 
    as.matrix()

rownames(raw_counts3) <- rownames(raw_counts2)


meta <- data.frame(metadata[-c(3 , 96),])
met <- data.frame(metrics[-c(3 , 96),])

full2 <- meta %>% left_join(met, by="sample")
rownames(full2) <- full2$sample

zipdata <- read.csv("/Users/emb016/Documents/HBC04292_Hazra/meta/zipcode_to_use.csv")

full3 <- full2 %>% left_join(zipdata, by=c("cc_study_id.y" = "Study_ID"))

## There are many zero heavy genes take a look at the number of zeros per gene
zero_dist <- rowSums(raw_counts3>0)

index <- which(zero_dist>116)
bad <- which(zero_dist<=116)
raw_counts4 <- raw_counts3[index,] 


vst <- vst(raw_counts4) 

genes_removed <- data.frame(rownames(raw_counts3[bad,]))
colnames(genes_removed) <- "gene"

genes_removed <- genes_removed %>% left_join(pruned_annotations, by=c("gene"="ensembl_gene_id"))

#write.table(genes_removed, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/genes_removed_for_zeros.csv", sep=",", row.names = FALSE)
```




## Scale continuous variables

Scaling is better for DESEQ2 and we scale age, intronic rate, Dx_FUtime, z_coi_nat, and percent GC.

We will model race, surgery type, and CaseControl as factors.

```{r}
full3$scaled_intronic <- scale(full3$intronic_rate)
full3$scaled_age <- scale(full3$Age_DCIS_Dx.x)
full3$scaled_FU <- scale(full3$Dx_FUtime.x)
full3$scaled_gc <- scale(full3$percent_gc)
full3$scaled_z_coi <- scale(full3$z_coi_nat)
full3$scaled_z_he <- scale(full3$z_he_nat)


full3$Race.x <- as.factor(full3$Race.x)
full3$Surgery_Type.x <- as.factor(full3$Surgery_Type.x)
full3$CaseControl.x <- as.factor(full3$CaseControl.x)



```




# DESEQ-   gene expression ~ Intronic Rate + Percent GC + Race  + Dx_FUtime +  Age + Grade + coi + coi*casecontrol.   - M1

**Note that we are modelling Grade as a continuous variable since grade 2 is intermediate to grade 1 and 3**

Differential gene expression analysis of count data was performed using the Bioconductor R package, DESeq2, which fits the count data to a negative binomial model.

Prior to identifying differentially expressed genes, the data was fit to the model and the dispersion was assessed to explore the fit of the count data to the model.

Using this model we will run DESEQ2. We often look at the dispersion plot to get a good idea of whether or not our data is a good fit for the model. Dispersion is a metric for variance which also takes into consideration mean expression. A dispersion value is estimated for each individual gene, then 'shrunken' to a more accurate value based on expected variation for the mean value. Finally, the shrunken dispersion value is used in the final GLM fit. From this plot we see the unshrunken (black) and shrunken (blue) dispersion values for every gene and that:

There is an expected inverse relationship between dispersion and mean expression. However, overall dispersions are very high. This is probably due to the high number of zeros. ***The data are an ok fit for the model.***


```{r}

## make sure metadata order matches count colnames

metadata2 <- data.frame(full3[match(full3$sample,colnames(raw_counts4)),])
rownames(metadata2) <- metadata2$sample

metadata2$Surgery_Type.x <- relevel(metadata2$Surgery_Type.x, ref="1")

dds <- DESeqDataSetFromMatrix(raw_counts4, 
                              colData = metadata2, 
                              design = ~  scaled_age + scaled_FU + scaled_gc + scaled_intronic + Race.x + Grade.x + scaled_z_coi + CaseControl.x*scaled_z_coi)


dds <- DESeq(dds)

# Dispersion plot
plotDispEsts(dds)
```



## Extract results 

Below we summarize the results for each contrast of interest. Significant DE genes are found in files with sig_genes in the title. Full results (p-values, log2 fold changes, adjusted p-values and normalized counts) for all genes tested for each contrast are found in files with all_results in the title.

### COI

Using a Wald test, we tested for significant differences related to age using an adjusted p-value cut-off of 0.1. Since we ran this as a continuous variable the log2fold changes are per unit of the variable so can seem small. 

There were 4 significant differentially expressed genes identified. 

### COI X CaseControl

Using a Wald test, we tested for an interaction between COI and case control status using an adjusted p-value cut-off of 0.1. 

There were 38 significantly DE genes identified.


```{r}

## get normalized counts
norm_c <- data.frame(counts(dds, normalized=TRUE))
norm_c$gene <- rownames(norm_c)

shrunken_res_coi <- data.frame(results(dds,
         name="scaled_z_coi"))



shrunken_res_int2 <- data.frame(results(dds,
         name="scaled_z_coi.CaseControl.x1"))



res_tb_coi <- shrunken_res_coi %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_coi <- res_tb_coi[res_tb_coi$baseMean>0,]


res_tb_int2 <- shrunken_res_int2 %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_int2 <- res_tb_int2[res_tb_int2$baseMean>0,]



# Subset to return genes with padj < 0.05



sig_genes_coi <- res_tb_coi %>%
        dplyr::filter(padj < 0.05) %>%
        arrange(padj)





sig_genes_int2 <- res_tb_int2 %>%
        dplyr::filter(padj < 0.05) %>%
        arrange(padj)


#write out results
 write.table(sig_genes_coi, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_coi_caseinteraction_p0.05.csv", sep=",", row.names = FALSE)
 write.table(res_tb_coi, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_coi_caseinteraction_p0.05.csv", sep=",", row.names = FALSE)
write.table(sig_genes_int2, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_caseinteraction_p0.05.csv", sep=",", row.names = FALSE)
 write.table(res_tb_int2, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_caseinteraction_p0.05.csv", sep=",", row.names = FALSE) 
 
# 
# write.table(sig_genes_FU, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_dxfutime_p0.1.csv", sep=",", row.names = FALSE)
# write.table(res_tb_FU, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_dxfutime_p0.1.csv", sep=",", row.names = FALSE)
# 
# 
# write.table(sig_genes_race, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_race_p0.1.csv", sep=",", row.names = FALSE)
# write.table(res_tb_race, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_race_p0.1.csv", sep=",", row.names = FALSE)
# 
# 
# write.table(res_tb_casecontrol, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_casecontrol_p0.1.csv", sep=",", row.names = FALSE)
# 
# write.table(sig_genes_grade, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_grade_p0.1.csv", sep=",", row.names = FALSE)
# write.table(res_tb_grade, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_grade_p0.1.csv", sep=",", row.names = FALSE)


```


### Top genes with COI

Here, we've plotted the abundance (DESeq2-normalized counts) of the top 20 differentially expressed genes (DEGs, ranked by adjusted p-value) in each sample. We plot COI vs. abundance and add a trendline for the top 4 genes.  Some of these look convincing but others do not.

**We plot this 2x, once zoomed in.**

```{r, fig.height=3, fig.width=3}

to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)
#Graph ones with a SIDS effect

rna = sig_genes_coi$gene
names = sig_genes_coi$external_gene_name

out <- vector("list", length = 4)
out2 <- vector("list", length = 4)

to_graph = data.frame(to_test[,rna])
to_graph$coi <- metadata2$scaled_z_coi


for (i in seq(1,4)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=coi,y=temp))  + geom_point()+ ylab(paste0(names[[i]])) + geom_smooth(method = "lm") + theme(legend.position = "NONE")
    
  out2[[i]] <- ggplot(to_graph,aes(x=coi,y=temp))  + geom_point()+ ylab(paste0(names[[i]])) + geom_smooth(method = "lm") + theme(legend.position = "NONE") + ylim(0,5000) + xlab("COI")

  
}

out
out2

```



### Top genes with Interaction between COI and Case/Control

Here, we've plotted the abundance (DESeq2-normalized counts) of the top 20 differentially expressed genes (DEGs, ranked by adjusted p-value) in each sample. We plot COI vs. abundance and add a trendline for the top 5 genes.  Some of these look convincing but others do not.

**We plot this 2x, once separated by race and once with both races together.**


```{r, fig.height=3, fig.width=8}


to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)
#Graph ones with a SIDS effect

rna = sig_genes_int2$gene
names = sig_genes_int2$external_gene_name

out <- vector("list", length = 7)
out2 <- vector("list", length = 7)

to_graph = data.frame(to_test[,rna])
to_graph$coi <- metadata2$scaled_z_coi
to_graph$casecontrol <- metadata2$CaseControl.x
to_graph$race <- metadata2$Race
to_graph$race <- gsub("1", "NHW", to_graph$race)
to_graph$race <- gsub("2", "Black", to_graph$race)
to_graph$race <- relevel(factor(to_graph$race), ref = "NHW")


for (i in seq(1,7)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=coi,y=temp, color=as.factor(casecontrol)))  + geom_point()+ ylab(paste0(names[[i]])) + geom_smooth(method = "lm") + xlab("COI") + scale_color_manual(values=c("#9ecae1","#3182bd")) + theme(legend.position = "None")

    out2[[i]] <- ggplot(to_graph,aes(x=coi,y=temp, color=as.factor(casecontrol)))  + geom_point()+ ylab(paste0(names[[i]])) + geom_smooth(method = "lm") + facet_wrap(~race) + xlab("COI") + scale_color_manual(values=c("#9ecae1","#3182bd")) + theme(legend.position = "None")
  
}

out
out2


```





# R session

List and version of tools used for the QC report generation.

```{r}
sessionInfo()
```