---
title: "Differential Expression"
author: "Emma Berdan"
date: "`r Sys.Date()`"
output:
   html_document:
      code_folding: hide
      df_print: paged
      highlights: pygments
      number_sections: true
      self_contained: true
      theme: default
      toc: true
      toc_float:
         collapsed: true
         smooth_scroll: true
---

# Overview

- Principal Investigator: Aditi Hazra
- Experiment: Breast Cancer patient samples

```{r, cache = FALSE, message = FALSE, warning=FALSE}
library(tidyverse)
library(knitr)
library(DESeq2)
library(DEGreport)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
library(biomaRt)
library(topGO)
library(KEGGREST)
library(Category)
library(ape)
library(data.table)
library(fgsea)
library(clusterProfiler)
library(sva)
library(gprofiler2)
library(janitor)
library(ggvenn)
library(ggplot2)
ggplot2::theme_set(theme_light(base_size = 14))
opts_chunk[["set"]](
    cache = FALSE,
    dev = c("png", "pdf"),
    error = TRUE,
    highlight = TRUE,
    message = FALSE,
    prompt = FALSE,
    tidy = FALSE,
    warning = FALSE,
    fig.height = 4)

sanitize_datatable = function(df, ...) {
 # remove dashes which cause wrapping
 DT::datatable(df, ..., rownames=gsub("-", "_", rownames(df)),
                   colnames=gsub("-", "_", colnames(df)))
}
```

# Load and pre-process data

We load our data. We look at how many zeros we have per gene and see that for many genes it is very high. We set a cutoff that at most 25 samples (from the full dataset) can have zero counts.


## Get annotations
```{r get gene_annotations}
 # library(biomaRt)
#

# Get gene annotations based on reference data version used for alignment/quantification from BiomaRt

# Specify the Ensembl release ## dataset used by bcbio (check archives if needed)
#
# # List current versions of datasets
#  ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL")
#  datasets <- listDatasets(ensembl)
# #
# # # Identify the proper archive to use for corresponding Ensembl release
#  archives <- listEnsemblArchives()
# #
# # # # Use ensembl GRCm39
#   ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL",
#                     dataset = "mmusculus_gene_ensembl")
# # #
# # # ## Build biomaRt query
#  filters = listFilters(ensembl)
#  attributes = listAttributes(ensembl)
# # #
#   gene_annotations <- getBM(attributes=c('ensembl_gene_id', 'entrezgene_id',  'external_gene_name', 'description'),
#         filters = 'ensembl_gene_id',
#        values = rownames(counts),
#         mart = ensembl)
# #
# # #remove duplicated ensembl ids with different entrez gene IDs.
#  pruned_annotations = gene_annotations %>%
#    arrange(ensembl_gene_id, ensembl_gene_id) %>%
#    distinct(ensembl_gene_id, .keep_all = TRUE)
# #
#  write.csv(pruned_annotations, "/Users/emb016/Documents/HBC04572_Rosen/data/biomart.csv")
#
 pruned_annotations <- read.csv("/Users/emb016/Documents/HBC04676_Sicinski/annotation/biomart.csv") %>% .[,-1]
```

## Load Data

```{r  show-metadata}
se <- readRDS("/Users/emb016/Documents/HBC04292_Hazra/full_data/bcbio-se.rds")

metadata <- colData(se) %>% 
    as_tibble(rownames = NULL)

metrics <- metadata(se)$metrics %>% 
    left_join(metadata, by = c("sample" = "sample"))

meta_sm <- metadata %>%
  as.data.frame() %>%
  column_to_rownames("sample")


raw_counts <- assays(se)[["raw"]] 



coding_genes <- read.table("/Users/emb016/Documents/HBC04292_Hazra/lnc_and_protein_hg38.txt", header=FALSE)

raw_counts2 <- subset(raw_counts, rownames(raw_counts) %in% coding_genes$V2) 



## Remove WU104 and WU228

raw_counts3 <- raw_counts2[,-c(3 , 96)] %>% as_tibble()  %>% 
    as.matrix()

rownames(raw_counts3) <- rownames(raw_counts2)


meta <- data.frame(metadata[-c(3 , 96),])
met <- data.frame(metrics[-c(3 , 96),])

full2 <- meta %>% left_join(met, by="sample")
rownames(full2) <- full2$sample



## There are many zero heavy genes take a look at the number of zeros per gene
zero_dist <- rowSums(raw_counts3>0)
hist(zero_dist)

index <- which(zero_dist>116)
bad <- which(zero_dist<=116)
raw_counts4 <- raw_counts3[index,] 


vst <- vst(raw_counts4) 

genes_removed <- data.frame(rownames(raw_counts3[bad,]))
colnames(genes_removed) <- "gene"

genes_removed <- genes_removed %>% left_join(pruned_annotations, by=c("gene"="ensembl_gene_id"))

write.table(genes_removed, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/genes_removed_for_zeros.csv", sep=",", row.names = FALSE)
```


## Look at samples by number of zeros

The number of zeros is a normal distribution and no factor seems to affect the distribution in a strong manner.

```{r}
zero_data <- data.frame(colnames(raw_counts3),colSums(raw_counts3==0))
colnames(zero_data) <- c("sample","zero_count")
hist(zero_data$zero_count)

## add more data

zero_data <- zero_data %>% left_join(metadata)

ggplot(zero_data, aes(x=CaseControl,y=zero_count,color=as.factor(Race))) + geom_boxplot() + geom_point(alpha=0.5,position = position_jitterdodge())


ggplot(zero_data, aes(x=Age_DCIS_Dx,y=zero_count,color=as.factor(Race))) +  geom_point(alpha=0.5) + geom_smooth(method="lm")

ggplot(zero_data, aes(x=Dx_FUtime,y=zero_count,color=as.factor(Race))) +  geom_point(alpha=0.5) + geom_smooth(method="lm")


```

# Subset to Ipsilateral

```{r}
ipsi <- subset(full2, full2$Recur_Loc.x != 1)

true_ipsi <- subset(full2, full2$cc_study_id.x %in% ipsi$cc_study_id.x)

raw_counts5 <- raw_counts4[,true_ipsi$sample] 


raw_counts6 <- raw_counts5


vst <- vst(raw_counts6) 
```



## Look at covariates

We only consider covariates that correlate with PC's explaining 3% or more of the variance for inclusion in the model. 

```{r,fig.height=12, fig.width=8}
full3 <- true_ipsi[,c(1,5,7,8,9,12,13,14,21,29, 31, 33, 34,35, 36,37,40 )] %>% data.frame()
rownames(full3) <- full3$sample


degCovariates(vst, metadata = full3, minPC=3)
```

## Scale continuous variables

Scaling is better for DESEQ2.

```{r}
full3$scaled_intronic <- scale(full3$intronic_rate)
full3$scaled_age <- scale(full3$Age_DCIS_Dx.x)
full3$scaled_FU <- scale(full3$Dx_FUtime.x)
full3$scaled_gc <- scale(full3$percent_gc)
full3$scaled_reads <- scale(full3$total_reads)
full3$Grade.x <- as.numeric(full3$Grade.x)


full3$Race.x <- as.factor(full3$Race.x)
full3$Grade.x <- as.factor(full3$Grade.x)
full3$CaseControl.x <- as.factor(full3$CaseControl.x)



```






# DESEQ -   gene expression ~ total_reads + Percent GC + Race  + Dx_FUtime +  Grade + case/control.   - M1

**Note that we are modelling Grade as a continuous variable since grade 2 is intermediate to grade 1 and 3**

Differential gene expression analysis of count data was performed using the Bioconductor R package, DESeq2, which fits the count data to a negative binomial model.

Prior to identifying differentially expressed genes, the data was fit to the model and the dispersion was assessed to explore the fit of the count data to the model.

Using this model we will run DESEQ2. We often look at the dispersion plot to get a good idea of whether or not our data is a good fit for the model. Dispersion is a metric for variance which also takes into consideration mean expression. A dispersion value is estimated for each individual gene, then 'shrunken' to a more accurate value based on expected variation for the mean value. Finally, the shrunken dispersion value is used in the final GLM fit. From this plot we see the unshrunken (black) and shrunken (blue) dispersion values for every gene and that:

There is an expected inverse relationship between dispersion and mean expression. However, overall dispersions are very high. This is probably due to the high number of zeros. ***The data are an ok fit for the model.***


```{r}

## make sure metadata order matches count colnames

metadata2 <- data.frame(full3[match(full3$sample,colnames(raw_counts6)),])
rownames(metadata2) <- metadata2$sample
metadata2$Grade.x <- as.numeric(metadata2$Grade.x)


dds <- DESeqDataSetFromMatrix(raw_counts6, 
                              colData = metadata2, 
                              design = ~  scaled_reads + scaled_FU + scaled_gc +  Race.x + Grade.x +  CaseControl.x)


dds <- DESeq(dds)

# Dispersion plot
plotDispEsts(dds)
```



## Extract results 

Below we summarize the results for each contrast of interest. Significant DE genes are found in files with sig_genes in the title. Full results (p-values, log2 fold changes, adjusted p-values and normalized counts) for all genes tested for each contrast are found in files with all_results in the title.


```{r}

## get normalized counts
norm_c <- data.frame(counts(dds, normalized=TRUE))
norm_c$gene <- rownames(norm_c)

test <- results(dds, name =  "CaseControl.x_1_vs_0")



shrunken_res_FU <- data.frame(lfcShrink(dds,
         coef="scaled_FU", type = "ashr"))

shrunken_res_race <- data.frame(lfcShrink(dds,
         coef="Race.x_2_vs_1", type = "ashr"))

shrunken_res_casecontrol <- data.frame(results(dds,
         name="CaseControl.x_1_vs_0"))


shrunken_res_grade <- data.frame(lfcShrink(dds,
         coef="Grade.x", type = "ashr"))



res_tb_FU <- shrunken_res_FU %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_FU <- res_tb_FU[res_tb_FU$baseMean>0,]


res_tb_race <- shrunken_res_race %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_race <- res_tb_race[res_tb_race$baseMean>0,]



res_tb_casecontrol <- shrunken_res_casecontrol %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_casecontrol <- res_tb_casecontrol[res_tb_casecontrol$baseMean>0,]




res_tb_grade <- shrunken_res_grade %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_grade <- res_tb_grade[res_tb_grade$baseMean>0,]



# Subset to return genes with padj < 0.05


sig_genes_race <- res_tb_race %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)

sig_genes_FU <- res_tb_FU %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)

sig_genes_casecontrol <- res_tb_casecontrol %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)


sig_genes_grade <- res_tb_grade %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)


#write out results
# write.table(sig_genes_age, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_age_p0.1.csv", sep=",", row.names = FALSE)
# write.table(res_tb_age, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_age_p0.1.csv", sep=",", row.names = FALSE)
# 
# write.table(sig_genes_FU, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_dxfutime_p0.1.csv", sep=",", row.names = FALSE)
 write.table(res_tb_FU, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_dxfutime_ipsilateral_only.csv", sep=",", row.names = FALSE)
# 
# 
# write.table(sig_genes_race, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_race_p0.1.csv", sep=",", row.names = FALSE)
 write.table(res_tb_race, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_race_ipsilateral_only.csv", sep=",", row.names = FALSE)
# 
# 
# write.table(res_tb_casecontrol, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_casecontrol_p0.1.csv", sep=",", row.names = FALSE)
# 
# write.table(sig_genes_grade, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_grade_p0.1.csv", sep=",", row.names = FALSE)
 write.table(res_tb_grade, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_grade_ipsilateral_only.csv", sep=",", row.names = FALSE)


```


## Volcano plots

Here, we plot the log2 foldchange of the genes against the log10 adjusted p-value. The points highlighted in red are genes that have padj < 0.1 and a log2 foldchange > 0.1. Points in blue have a padj < 0.1 and a log2 foldchange < 0.1 and points in green have a padj > 0.1 and a log2 foldchange > 0.1. Grey points are non-significant. The dashed lines correspond to the cutoff values of log2 foldchance and padj that we have chosen.

We see plot this for all of our main contrasts. 

```{r volcano-plots yumm m1, fig.align="center",fig.height=5, fig.width=6.5}
library("EnhancedVolcano")
EnhancedVolcano(res_tb_casecontrol,
                lab= res_tb_casecontrol$external_gene_name, pCutoff = 0.1
, selectLab = c(sig_genes_casecontrol$external_gene_name[1:15]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'padj', title="Case vs. Control", subtitle = "", drawConnectors = T)

EnhancedVolcano(res_tb_FU,
                lab= res_tb_FU$external_gene_name, pCutoff = 0.1
, selectLab = c(sig_genes_FU$external_gene_name[1:15]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'padj', title="Dx.FUtime", subtitle = "", drawConnectors = T)

#pdf("/Users/emb016/Documents/HBC04292_Hazra/full_data/race_volcano_ipsilateral.pdf", width = 10, height = 6)
EnhancedVolcano(res_tb_race,
                lab= res_tb_race$external_gene_name, pCutoff = 0.1
, selectLab = c(sig_genes_race$external_gene_name[1:15]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'padj', title="Race", subtitle = "", drawConnectors = T) + xlim(-3,3)
#dev.off()

EnhancedVolcano(res_tb_grade,
                lab= res_tb_grade$external_gene_name, pCutoff = 0.1
, selectLab = c(sig_genes_grade$external_gene_name[1:15]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'padj', title="Grade", subtitle = "", drawConnectors = T)




```

### Top genes with Race

Here, we've plotted the abundance (DESeq2-normalized counts) of the top 15 differentially expressed genes (DEGs, ranked by adjusted p-value) in each sample. We plot Race and case/control vs. abundance. Some of these look convincing but others do not.**Control samples (0) are in red and case samples (1) are in blue**

```{r, fig.height=4, fig.width=4}

to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)

rna = sig_genes_race$gene
names = sig_genes_race$external_gene_name

out <- vector("list", length = 15)

to_graph = data.frame(to_test[,rna])
to_graph$race <- metadata2$Race.x

to_graph$race <- gsub("1", "NHW", to_graph$race)
to_graph$race <- gsub("2", "Black", to_graph$race)

to_graph$case_control <- metadata2$CaseControl.x

for (i in seq(1,15)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=race,y=temp, color=case_control)) + geom_boxplot() + geom_point(alpha=0.5,position = position_jitterdodge())+  ylab(paste0(names[[i]])) + geom_smooth() + theme(legend.position = "NONE") + xlab("Race")

  
}

out




```

### Top genes with Grade

Here, we've plotted the abundance (DESeq2-normalized counts) of the top 10 differentially expressed genes (DEGs, ranked by adjusted p-value) in each sample. We plot Grade vs. abundance. Some of these look convincing but others do not. 




```{r, fig.height=3, fig.width=3}

to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)
#Graph ones with a SIDS effect

rna = sig_genes_grade$gene
names = sig_genes_grade$external_gene_name

out <- vector("list", length = 10)

to_graph = data.frame(to_test[,rna])
to_graph$grade <- as.factor(metadata2$Grade.x)
to_graph$case_control <- metadata2$CaseControl.x


for (i in seq(1,10)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=grade,y=temp)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[i]])) + geom_smooth() + theme(legend.position = "NONE")

  
}

out

```


### Top genes with Case/Control

Here, weâ€™ve plotted the abundance (DESeq2-normalized counts) of the 3 differentially expressed genes in each sample.


```{r, fig.height=3, fig.width=3}


genes <- c("RSAD2","IFI27","IL7","USP18","HERC6","LAMP3","EPSTI1","OASL")
sub <- sig_genes_casecontrol
to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)


rna = sub$gene[1:3]
names = sub$external_gene_name[1:3]

out <- vector("list", length = 3)
out2 <- vector("list", length = 3)
out3 <- vector("list", length = 3)


to_graph = data.frame(to_test[,rna])
to_graph$case_control <- metadata2$CaseControl.x

for (i in seq(1,3)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=case_control,y=temp)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[i]])) + geom_smooth() + theme(legend.position = "NONE")



#    out3[[i]] <- ggplot(to_graph,aes(x=case_control,y=temp)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[i]])) + geom_smooth() + theme(legend.position = "NONE") + ylim(0,500)
  
}

out

#out3
```


# Figure 2

Code to create figure 2 from the manuscript
```{r, fig.height=8, fig.width=10}
volcano <- EnhancedVolcano(res_tb_casecontrol,
                lab= res_tb_casecontrol$external_gene_name, pCutoff = 0.1
, selectLab = c(sig_genes_casecontrol$external_gene_name[1:15]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'padj', subtitle = "", drawConnectors = T)

#chgb  <- ggplot(to_graph,aes(x=case_control,y=ENSG00000089199, color=case_control)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[1]])) + geom_smooth() + theme(legend.position = "NONE") + scale_color_manual(values=c("#fc9272","#de2d26")) + xlab("Case/Control") + scale_x_discrete(labels=c('Control', 'Case'))

rmb20 <-ggplot(to_graph,aes(x=case_control,y=ENSG00000203867, color=case_control)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[1]])) + geom_smooth() + theme(legend.position = "NONE") + scale_color_manual(values=c("#fc9272","#de2d26")) + xlab("Case/Control") + scale_x_discrete(labels=c('Control', 'Case'))

#syp <-ggplot(to_graph,aes(x=case_control,y=ENSG00000102003, color=case_control)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[2]])) + geom_smooth() + theme(legend.position = "NONE") + scale_color_manual(values=c("#fc9272","#de2d26")) + xlab("Case/Control") + scale_x_discrete(labels=c('Control', 'Case'))

synj <- ggplot(to_graph,aes(x=case_control,y=ENSG00000213463, color=case_control)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[2]])) + geom_smooth() + theme(legend.position = "NONE") + scale_color_manual(values=c("#fc9272","#de2d26")) + xlab("Case/Control") + scale_x_discrete(labels=c('Control', 'Case'))

slc <- ggplot(to_graph,aes(x=case_control,y=ENSG00000080493, color=case_control)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[3]])) + geom_smooth() + theme(legend.position = "NONE") + scale_color_manual(values=c("#fc9272","#de2d26")) + xlab("Case/Control") + scale_x_discrete(labels=c('Control', 'Case'))



library(patchwork)
volcano / (rmb20|synj|slc) +
  plot_annotation(tag_levels = 'a', tag_suffix=")")
      
pdf("/Users/emb016/Documents/HBC04292_Hazra/full_data/Fig2_ipsilateral.pdf", width = 8, height = 8)
volcano / (rmb20|synj|slc) +
  plot_annotation(tag_levels = 'a', tag_suffix=")")
dev.off()

```


## Race Figure (figure 8)

Code to create Figure 8 from the manuscript.
```{r}
 
to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)

rna = sig_genes_race$gene
names = sig_genes_race$external_gene_name

out <- vector("list", length = 15)

to_graph = data.frame(to_test[,rna])
to_graph$race <- metadata2$Race.x

to_graph$race <- gsub("1", "NHW", to_graph$race)
to_graph$race <- gsub("2", "Black", to_graph$race)

to_graph$race <- relevel(as.factor(to_graph$race), ref="NHW")

to_graph$case_control <- metadata2$CaseControl.x

race = ggplot(to_graph,aes(x=race,y=ENSG00000238083, color=case_control)) + geom_boxplot() + geom_point(alpha=0.5,position=position_jitterdodge())+  ylab(paste0(names[[1]])) + geom_smooth() + theme(legend.position = "NONE") + scale_color_manual(values=c("#efedf5",
"#756bb1")) + xlab("Race") 


ggsave(file = "/Users/emb016/Documents/HBC04292_Hazra/full_data/Fig8_Race_ipsi.png", race)  ## save plot

```




## Enrichment for Case/Control

 Here we indicate the directionality of the change and use 
 [Gene Set Enrichment Analysis (GSEA)](https://www.gsea-msigdb.org/gsea/index.jsp). GSEA aggregates the per gene statistics across genes within a gene set, therefore making it possible to detect situations where all genes in a predefined set change in a small but coordinated way. Since it is likely that many relevant phenotypic differences are manifested by small but consistent changes in a set of genes.

**We will test enrichment using the msigdb hallmark gene set. For each test if we have results we will also make a dotplot. The dotplot shows the gene ratio for each term ((count of core enrichment genes) / (count of pathway genes)). The size of the circle indicates the number of genes and the color indicates the p-value.**

## Load and format msigdb gene sets

```{r}

# The gene set was downloaded on June 12 2023
hallmark = read.table("/Users/emb016/Documents/HBC04592_Baecher_Allan/data/h.all.v2023.1.Hs.symbols.gmt", sep="\t", fill=TRUE)

#remove urls
hallmark = hallmark[,-2]


hallmark_melt = melt(hallmark, id=c("V1"))

## get rid of middle column

hallmark_melt = hallmark_melt[,c(1,3)]

colnames(hallmark_melt) = c("pathway_id","gene")


### prepare reactome full db
react = read.delim("/Users/emb016/Documents/HBC04592_Baecher_Allan/data/reactome_all.txt", header=FALSE)  %>% dplyr::filter(V6 == "Homo sapiens") %>%
  dplyr::select(c(V1, V2, V4))
colnames(react) <- c("gene", "pathway_id", "pathway")


## remove transcripts

react <- react[grep("ENSG", react$gene), ]

```

### Case vs. Control

#### Prepare the rankings
- based on the logFCs

```{r}
# Prepare the gene list

## For each gene just keep the closest peak

results_entrez <- res_tb_casecontrol %>% arrange(desc(log2FoldChange))
logFCs <- pull(results_entrez, log2FoldChange)
names(logFCs) <- pull(results_entrez, external_gene_name)
```



#### Run GSEA with MsigDB hallmark database 
```{r, fig.height=5, fig.width=8}
gsea_all <- GSEA(geneList = logFCs,
                 TERM2GENE = hallmark_melt,
                 pvalueCutoff = 0.05,
                 pAdjustMethod = "BH",
                 verbose = TRUE,
                 seed = TRUE)


gsea_tbl <- gsea_all %>%
    slot("result") %>%
    as_tibble() 

clusterProfiler::dotplot(gsea_all, title="Hallmark Enrichment Case vs. Control")

DT::datatable(gsea_tbl)

pdf("/Users/emb016/Documents/HBC04292_Hazra/full_data/hallmark_ipsilateral.pdf", width = 8, height = 6)
clusterProfiler::dotplot(gsea_all, title="Hallmark Enrichment Case vs. Control")

dev.off()

```



### Run GSEA with reactome
```{r, fig.width=8}
## redo ranking with ensembl ids

results_entrez <- res_tb_casecontrol %>% arrange(desc(log2FoldChange))
logFCs_react <- pull(results_entrez, log2FoldChange)
names(logFCs_react) <- pull(results_entrez, gene)



react2 <- react[,c(3,1)]
gsea_all2 <- GSEA(geneList = logFCs_react,
                 TERM2GENE = react2,
                 pvalueCutoff = 0.1,
                 pAdjustMethod = "BH",
                 verbose = TRUE,
                 seed = TRUE)


gsea_tbl2 <- gsea_all2 %>%
    slot("result") %>%
    as_tibble() 

DT::datatable(gsea_tbl2)


pdf("/Users/emb016/Documents/HBC04292_Hazra/full_data/reactome_ipsilateral.pdf", width = 8, height = 6)
clusterProfiler::dotplot(gsea_all2, title="Reactome Enrichment Case vs. Control")
dev.off()

#clusterProfiler::dotplot(gsea_all2, title="Reactome Enrichment in Case vs. Control")

```



# R session

List and version of tools used for the QC report generation.

```{r}
sessionInfo()
```