---
title: "Differential Expression"
author: "Emma Berdan"
date: "`r Sys.Date()`"
output:
   html_document:
      code_folding: hide
      df_print: paged
      highlights: pygments
      number_sections: true
      self_contained: true
      theme: default
      toc: true
      toc_float:
         collapsed: true
         smooth_scroll: true
---

# Overview

- Principal Investigator: Aditi Hazra
- Experiment: Breast Cancer patient samples

```{r, cache = FALSE, message = FALSE, warning=FALSE}
library(tidyverse)
library(knitr)
library(DESeq2)
library(DEGreport)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
library(biomaRt)
library(topGO)
library(KEGGREST)
library(Category)
library(ape)
library(data.table)
library(fgsea)
library(clusterProfiler)
library(sva)
library(gprofiler2)
library(janitor)
library(ggvenn)
library(ggplot2)
library(MASS)
library(reshape2)
ggplot2::theme_set(theme_light(base_size = 14))
opts_chunk[["set"]](
    cache = FALSE,
    dev = c("png", "pdf"),
    error = TRUE,
    highlight = TRUE,
    message = FALSE,
    prompt = FALSE,
    tidy = FALSE,
    warning = FALSE,
    fig.height = 4)

sanitize_datatable = function(df, ...) {
 # remove dashes which cause wrapping
 DT::datatable(df, ..., rownames=gsub("-", "_", rownames(df)),
                   colnames=gsub("-", "_", colnames(df)))
}
```

# Load and pre-process data

We load our data. We look at how many zeros we have per gene and see that for many genes it is very high. We set a cutoff that at most 25 samples can have zero counts.


## Get annotations
```{r get gene_annotations}
 # library(biomaRt)
#

# Get gene annotations based on reference data version used for alignment/quantification from BiomaRt

# Specify the Ensembl release ## dataset used by bcbio (check archives if needed)
#
# # List current versions of datasets
#  ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL")
#  datasets <- listDatasets(ensembl)
# #
# # # Identify the proper archive to use for corresponding Ensembl release
#  archives <- listEnsemblArchives()
# #
# # # # Use ensembl GRCm39
#   ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL",
#                     dataset = "mmusculus_gene_ensembl")
# # #
# # # ## Build biomaRt query
#  filters = listFilters(ensembl)
#  attributes = listAttributes(ensembl)
# # #
#   gene_annotations <- getBM(attributes=c('ensembl_gene_id', 'entrezgene_id',  'external_gene_name', 'description'),
#         filters = 'ensembl_gene_id',
#        values = rownames(counts),
#         mart = ensembl)
# #
# # #remove duplicated ensembl ids with different entrez gene IDs.
#  pruned_annotations = gene_annotations %>%
#    arrange(ensembl_gene_id, ensembl_gene_id) %>%
#    distinct(ensembl_gene_id, .keep_all = TRUE)
# #
#  write.csv(pruned_annotations, "/Users/emb016/Documents/HBC04572_Rosen/data/biomart.csv")
#
 pruned_annotations <- read.csv("/Users/emb016/Documents/HBC04676_Sicinski/annotation/biomart.csv") %>% .[,-1]
```

## Load Data

```{r  show-metadata}
se <- readRDS("/Users/emb016/Documents/HBC04292_Hazra/full_data/bcbio-se.rds")

metadata <- colData(se) %>% 
    as_tibble(rownames = NULL)

metrics <- metadata(se)$metrics %>% 
    left_join(metadata, by = c("sample" = "sample"))

meta_sm <- metadata %>%
  as.data.frame() %>%
  column_to_rownames("sample")


raw_counts <- assays(se)[["raw"]] 



coding_genes <- read.table("/Users/emb016/Documents/HBC04292_Hazra/lnc_and_protein_hg38.txt", header=FALSE)

raw_counts2 <- subset(raw_counts, rownames(raw_counts) %in% coding_genes$V2) 



## Remove WU104 and WU228

raw_counts3 <- raw_counts2[,-c(3 , 96)] %>% as_tibble()  %>% 
    as.matrix()

rownames(raw_counts3) <- rownames(raw_counts2)


meta <- data.frame(metadata[-c(3 , 96),])
met <- data.frame(metrics[-c(3 , 96),])

full2 <- meta %>% left_join(met, by="sample")
rownames(full2) <- full2$sample



## There are many zero heavy genes take a look at the number of zeros per gene
zero_dist <- rowSums(raw_counts3>0)
hist(zero_dist)

index <- which(zero_dist>116)
bad <- which(zero_dist<=116)
raw_counts4 <- raw_counts3[index,] 


vst <- vst(raw_counts4) 

genes_removed <- data.frame(rownames(raw_counts3[bad,]))
colnames(genes_removed) <- "gene"



## load cibersort

ciber <- read.csv("/Users/emb016/Documents/HBC04292_Hazra/CIBERSORTx_Job1_Results.csv")

ciber_sub <- ciber[,1:23]

meta_good <- meta[,c(1,5,7,9,14,29)]

ciber_sub <- ciber_sub %>% left_join(meta_good, by=c("Mixture"="sample"))

ciber_melt <- melt(ciber_sub,id=c("Mixture","Age_DCIS_Dx","CaseControl","Dx_FUtime","Grade","Race"))
```


## Graph Cibersort Results

```{r, fig.width=10, fig.height=7}

ggplot(ciber_melt, aes(x=variable,y=value,color=as.factor(Race))) + geom_boxplot() + geom_point(alpha=0.5,position = position_jitterdodge()) +  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + ylab("Proportion") + xlab("Cell Type")

ggplot(ciber_melt, aes(x=variable,y=value,color=as.factor(CaseControl))) + geom_boxplot() + geom_point(alpha=0.5,position = position_jitterdodge()) +  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + ylab("Proportion") + xlab("Cell Type") +  scale_color_manual(values=c("#bcbddc", "#4a1486")) 


ggplot(ciber_melt, aes(x=variable,y=value,color=as.factor(Grade))) + geom_boxplot() + geom_point(alpha=0.5,position = position_jitterdodge()) +  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + ylab("Proportion") + xlab("Cell Type")

```

```{r, fig.width=15, fig.height=7}

ggplot(ciber_melt, aes(x=variable,y=value,color=as.factor(CaseControl))) + geom_boxplot() + geom_point(alpha=0.5,position = position_jitterdodge()) +  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + ylab("Proportion") + xlab("Cell Type") + facet_wrap(~Race) +



```


## Look at covariates

We input proportions and it looks like only Dx.FUtime has an impact on proportion

```{r,fig.height=12, fig.width=8}

props <- ciber[,1:23]
rownames(props) <- props$Mixture

props <- props[,-1]

props2 <- t(props)

met <- meta_good
rownames(met) <- met$sample
degCovariates(props2, metadata = met, minPC=3)
```

## Graph DX FU time

```{r}


out <- vector("list", length = 23)

rownames(ciber_sub) <- ciber_sub$Mixture
ciber_sub2 <- ciber_sub[,-1]
names <- colnames(ciber_sub2)[1:23]

cols <-data

to_graph <- ciber_sub2

for (i in seq(1,22)) {
  
  to_graph$temp=ciber_sub2[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=Dx_FUtime,y=temp))  + geom_point(alpha=0.5)+  ylab(paste0(names[[i]])) + geom_smooth(method = "lm") 

  
}

out

```




# R session

List and version of tools used for the QC report generation.

```{r}
sessionInfo()
```