---
title: "Differential Expression"
author: "Emma Berdan"
date: "`r Sys.Date()`"
output:
   html_document:
      code_folding: hide
      df_print: paged
      highlights: pygments
      number_sections: true
      self_contained: true
      theme: default
      toc: true
      toc_float:
         collapsed: true
         smooth_scroll: true
---

# Overview

- Principal Investigator: Aditi Hazra
- Experiment: Breast Cancer patient samples

```{r, cache = FALSE, message = FALSE, warning=FALSE}
library(tidyverse)
library(knitr)
library(DESeq2)
library(DEGreport)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
library(biomaRt)
library(org.Mm.eg.db)
library(topGO)
library(KEGGREST)
library(Category)
library(ape)
library(data.table)
library(fgsea)
library(clusterProfiler)
library(sva)
library(gprofiler2)
library(janitor)
library(ggvenn)
library(ggplot2)
ggplot2::theme_set(theme_light(base_size = 14))
opts_chunk[["set"]](
    cache = FALSE,
    dev = c("png", "pdf"),
    error = TRUE,
    highlight = TRUE,
    message = FALSE,
    prompt = FALSE,
    tidy = FALSE,
    warning = FALSE,
    fig.height = 4)

sanitize_datatable = function(df, ...) {
 # remove dashes which cause wrapping
 DT::datatable(df, ..., rownames=gsub("-", "_", rownames(df)),
                   colnames=gsub("-", "_", colnames(df)))
}
```

# Load and pre-process data

We load our data. We look at how many zeros we have per gene and see that for many genes it is very high. We set a cutoff that at least 25 samples must have non-zero counts.

```{r  show-metadata}
se <- readRDS("/Users/emb016/Documents/HBC04292_Hazra/full_data/bcbio-se.rds")

metadata <- colData(se) %>% 
    as_tibble(rownames = NULL)

metrics <- metadata(se)$metrics %>% 
    left_join(metadata, by = c("sample" = "sample"))

meta_sm <- metadata %>%
  as.data.frame() %>%
  column_to_rownames("sample")


raw_counts <- assays(se)[["raw"]] 



coding_genes <- read.table("/Users/emb016/Documents/HBC04292_Hazra/lnc_and_protein_hg38.txt", header=FALSE)

raw_counts2 <- subset(raw_counts, rownames(raw_counts) %in% coding_genes$V2) 



## Remove WU104 and WU228

raw_counts3 <- raw_counts2[,-c(3 , 96)] %>% as_tibble()  %>% 
    as.matrix()

rownames(raw_counts3) <- rownames(raw_counts2)


meta <- data.frame(metadata[-c(3 , 96),])
met <- data.frame(metrics[-c(3 , 96),])

full2 <- meta %>% left_join(met, by="sample")
rownames(full2) <- full2$sample



## There are many zero heavy genes take a look at the number of zeros per gene
zero_dist <- rowSums(raw_counts3>0)
hist(zero_dist)

index <- which(zero_dist>116)
raw_counts4 <- raw_counts3[index,] 


vst <- vst(raw_counts4) 
```

## Look at covariates

We only consider covariates that correlate with PC's explaining 3% or more of the variance for inclusion in the model. These are: Percent GC, Intronic Rate, Grade, Dx_FUtime, Age, and surgery type. 

**As intronic and exonic rate are usually correlated we will just use intronic rate. As age and menopause are correlated we will use age. **

```{r,fig.height=12, fig.width=8}
full3 <- full2[,c(1,2,5,7,8,9,12,13,14,15,19,21,29,30,31,34,35,36,37)]

rownames(full3) <- full3$sample


degCovariates(vst, metadata = full3, minPC=3)
```

## Scale continuous variables

Scaling is better for DESEQ2 and we scale age, intronic rate, Dx_FUtime, and percent GC.

We will model race,  surgery type, and CaseControl as factors.

```{r}
full3$scaled_intronic <- scale(full3$intronic_rate)
full3$scaled_age <- scale(full3$Age_DCIS_Dx.x)
full3$scaled_FU <- scale(full3$Dx_FUtime.x)
full3$scaled_gc <- scale(full3$percent_gc)

full3$Race.x <- as.factor(full3$Race.x)
full3$Surgery_Type.x <- as.factor(full3$Surgery_Type.x)
full3$CaseControl.x <- as.factor(full3$CaseControl.x)



```



## Get annotations
```{r get gene_annotations}
 # library(biomaRt)
#

# Get gene annotations based on reference data version used for alignment/quantification from BiomaRt

# Specify the Ensembl release ## dataset used by bcbio (check archives if needed)
#
# # List current versions of datasets
#  ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL")
#  datasets <- listDatasets(ensembl)
# #
# # # Identify the proper archive to use for corresponding Ensembl release
#  archives <- listEnsemblArchives()
# #
# # # # Use ensembl GRCm39
#   ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL",
#                     dataset = "mmusculus_gene_ensembl")
# # #
# # # ## Build biomaRt query
#  filters = listFilters(ensembl)
#  attributes = listAttributes(ensembl)
# # #
#   gene_annotations <- getBM(attributes=c('ensembl_gene_id', 'entrezgene_id',  'external_gene_name', 'description'),
#         filters = 'ensembl_gene_id',
#        values = rownames(counts),
#         mart = ensembl)
# #
# # #remove duplicated ensembl ids with different entrez gene IDs.
#  pruned_annotations = gene_annotations %>%
#    arrange(ensembl_gene_id, ensembl_gene_id) %>%
#    distinct(ensembl_gene_id, .keep_all = TRUE)
# #
#  write.csv(pruned_annotations, "/Users/emb016/Documents/HBC04572_Rosen/data/biomart.csv")
#
 pruned_annotations <- read.csv("/Users/emb016/Documents/HBC04676_Sicinski/annotation/biomart.csv") %>% .[,-1]
```


# DESEQ -   gene expression ~ Intronic Rate + Percent GC + Race  + Dx_FUtime + Grade + Age + surgery type + case/control.   - M1

**Since all except 2 individuals have type 1 or 2 surgery we will set type 1 as the base to pull out the 1 vs. 2 comparison** 

Differential gene expression analysis of count data was performed using the Bioconductor R package, DESeq2, which fits the count data to a negative binomial model.

Prior to identifying differentially expressed genes, the data was fit to the model and the dispersion was assessed to explore the fit of the count data to the model.

Using this model we will run DESEQ2. We often look at the dispersion plot to get a good idea of whether or not our data is a good fit for the model. Dispersion is a metric for variance which also takes into consideration mean expression. A dispersion value is estimated for each individual gene, then 'shrunken' to a more accurate value based on expected variation for the mean value. Finally, the shrunken dispersion value is used in the final GLM fit. From this plot we see the unshrunken (black) and shrunken (blue) dispersion values for every gene and that:

There is an expected inverse relationship between dispersion and mean expression. However, overall dispersions are very high. This is probably due to the high number of zeros. ***The data are an ok fit for the model.***


```{r}

## make sure metadata order matches count colnames

metadata2 <- data.frame(full3[match(full3$sample,colnames(raw_counts4)),])
rownames(metadata2) <- metadata2$sample

metadata2$Surgery_Type.x <- relevel(metadata2$Surgery_Type.x, ref="1")

dds <- DESeqDataSetFromMatrix(raw_counts4, 
                              colData = metadata2, 
                              design = ~  scaled_age + scaled_FU + scaled_gc + scaled_intronic + Race.x + Surgery_Type.x + Grade.x + CaseControl.x)


dds <- DESeq(dds)

# Dispersion plot
plotDispEsts(dds)
```



## Extract results 

Below we summarize the results for each contrast of interest. Significant DE genes are found in files with sig_genes in the title. Full results (p-values, log2 fold changes, adjusted p-values and normalized counts) for all genes tested for each contrast are found in files with all_results in the title.

### Age

Using a Wald test, we tested for significant differences related to age using an adjusted p-value cut-off of 0.1. Since we ran this as a continuous variable the log2fold changes are per unit of the variable so can seem small. **Positive logFC means that expression is higher in older individuals and a negative value means that it is higher in younger individuals.** 

There were 148 significant differentially expressed genes identified, of these genes 68 were up-regulated and 80 were down-regulated. 

### Case vs. control

Using a Wald test, we tested for significant differences between cases and controls using an adjusted p-value cut-off of 0.1. Nothing was significant here. The lowest adjusted p-value we have is 0.27.


### Dx.FUtime

Using a Wald test, we tested for significant differences related to Dx.FUtime using an adjusted p-value cut-off of 0.1. Since we ran this as a continuous variable the log2fold changes are per unit of the variable so can seem small. **Positive logFC means that expression is higher with larger values of Dx.FUtime and a negative value means that it is higher with smaller values of Dx.FUtime.** 

There were 532 significant differentially expressed genes identified, of these genes 172 were up-regulated and 354 were down-regulated. 


### Race

Using a Wald test, we tested for significant differences between scar cells and normal skin cells using an adjusted p-value cut-off of 0.1. The log2 fold changes are for the race2 samples relative to race1 samples. **Therefore, positive logFC means that expression is higher in the race 2 than race 1.** 

There were 69 significant differentially expressed genes identified, of these genes 38 were up-regulated and 31 were down-regulated. 


### Surgery type 2 vs. 1

Using a Wald test, we tested for significant differences between surgery type 1 and surgery type 2 using an adjusted p-value cut-off of 0.1. The log2 fold changes are for the surgery type 2 samples relative to surgery type 2 samples. **Therefore, positive logFC means that expression is higher in the surgery type2 than surgery type1.** 

There were 40 significant differentially expressed genes identified, of these genes 17 were up-regulated and 23 were down-regulated. 





```{r}

## get normalized counts
norm_c <- data.frame(counts(dds, normalized=TRUE))
norm_c$gene <- rownames(norm_c)



shrunken_res_age <- lfcShrink(dds,
         coef = "scaled_age", 
          type = "apeglm")

shrunken_res_FU <- lfcShrink(dds,
         coef = "scaled_FU", 
          type = "apeglm")

shrunken_res_race <- lfcShrink(dds,
         coef = "Race.x_2_vs_1", 
          type = "apeglm")

shrunken_res_casecontrol <- lfcShrink(dds,
         coef = "CaseControl.x_1_vs_0", 
          type = "apeglm")


shrunken_res_grade <- lfcShrink(dds,
         coef = "Grade.x", 
          type = "apeglm")

shrunken_res_surgery <- lfcShrink(dds,
         coef = "Surgery_Type.x_2_vs_1", 
          type = "apeglm")


# Create a results tibble for treatment
res_tb_age <- shrunken_res_age %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_age <- res_tb_age[res_tb_age$baseMean>0,]

res_tb_FU <- shrunken_res_FU %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_FU <- res_tb_FU[res_tb_FU$baseMean>0,]


res_tb_race <- shrunken_res_race %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_race <- res_tb_race[res_tb_race$baseMean>0,]



res_tb_casecontrol <- shrunken_res_casecontrol %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_casecontrol <- res_tb_casecontrol[res_tb_casecontrol$baseMean>0,]


res_tb_grade <- shrunken_res_grade %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_grade <- res_tb_grade[res_tb_grade$baseMean>0,]


res_tb_surgery <- shrunken_res_surgery %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_surgery <- res_tb_surgery[res_tb_surgery$baseMean>0,]





# Subset to return genes with padj < 0.05

sig_genes_age <- res_tb_age %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)

sig_genes_race <- res_tb_race %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)

sig_genes_FU <- res_tb_FU %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)

sig_genes_casecontrol <- res_tb_casecontrol %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)


sig_genes_surgery <- res_tb_surgery %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)



#write out results
write.table(sig_genes_age, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_age_p0.1.csv", sep=",", row.names = FALSE)
write.table(res_tb_age, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_age_p0.1.csv", sep=",", row.names = FALSE)

write.table(sig_genes_FU, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_dxfutime_p0.1.csv", sep=",", row.names = FALSE)
write.table(res_tb_FU, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_dxfutime_p0.1.csv", sep=",", row.names = FALSE)


write.table(sig_genes_race, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_race_p0.1.csv", sep=",", row.names = FALSE)
write.table(res_tb_race, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_race_p0.1.csv", sep=",", row.names = FALSE)


write.table(sig_genes_surgery, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_surgery_p0.1.csv", sep=",", row.names = FALSE)
write.table(res_tb_surgery, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_surgery_p0.1.csv", sep=",", row.names = FALSE)


```


## Volcano plots

Here, we plot the log2 foldchange of the genes against the log10 adjusted p-value. The points highlighted in red are genes that have padj < 0.1 and a log2 foldchange > 0.1. Points in blue have a padj < 0.1 and a log2 foldchange < 0.1 and points in green have a padj > 0.1 and a log2 foldchange > 0.1. Grey points are non-significant. The dashed lines correspond to the cutoff values of log2 foldchance and padj that we have chosen.

We see plot this for all of our main contrasts. 

```{r volcano-plots yumm m1, fig.align="center",fig.height=5, fig.width=6.5}
library("EnhancedVolcano")
EnhancedVolcano(res_tb_age,
                lab= res_tb_age$external_gene_name, pCutoff = 0.0010848211
, selectLab = c(sig_genes_age$external_gene_name[1:15]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'pvalue', title="Age", subtitle = "", drawConnectors = T, xlim=c(-1.5,1.5))

EnhancedVolcano(res_tb_FU,
                lab= res_tb_FU$external_gene_name, pCutoff = 0.004125941
, selectLab = c(sig_genes_FU$external_gene_name[1:15]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'pvalue', title="Dx.FUtime", subtitle = "", drawConnectors = T, xlim=c(-1,1))


EnhancedVolcano(res_tb_race,
                lab= res_tb_race$external_gene_name, pCutoff = 7.279255e-04
, selectLab = c(sig_genes_race$external_gene_name[1:15]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'pvalue', title="Race", subtitle = "", drawConnectors = T, xlim=c(-3,3))


EnhancedVolcano(res_tb_surgery,
                lab= res_tb_surgery$external_gene_name, pCutoff = 2.902991e-04
, selectLab = c(sig_genes_surgery$external_gene_name[1:15]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'pvalue', title="Surgery Type 1 vs. 0", subtitle = "", drawConnectors = T, xlim=c(-2,2))



```


## Heatmap of expression - Age

This plot shows only differentially expressed genes (padj < 0.1) on a per-sample basis. We have scaled the data by row and used the `ward.D2` method for clustering. We hope to see our differentially expressed genes cluster by sample group with similar expression patterns across all replicates, which they sort of do!

```{r heatmap-de age, fig.height=6}
# Extract significant genes
sigGenes <- sig_genes_age$gene

### Extract normalized expression for significant genes 
normalized_counts <- counts(dds, normalized=TRUE)
norm_sig <- normalized_counts[sigGenes,] 
meta <- metadata2$Age_DCIS_Dx.x %>% data.frame()
rownames(meta) <- colnames(norm_sig)
### Set a color palette
heat_colors <- colorRampPalette(c("blue", "black", "yellow"))(100)
anno_colors <- list(condition=c(KO="#20A387FF",control="#453788FF"))

### Run pheatmap using the metadata data frame for the annotation (11 x 5)
pheatmap(norm_sig, 
    color = heat_colors, 
    cluster_rows = T, 
    show_rownames = F,
    annotation = meta, 
    annotation_colors = anno_colors,
    border_color = NA, 
    fontsize = 10, 
    scale = "row", 
    fontsize_row = 10, 
    height = 20)
```
## Heatmap of expression - Dx.FUtime

This plot shows only differentially expressed genes (padj < 0.1) on a per-sample basis. We have scaled the data by row and used the `ward.D2` method for clustering. We hope to see our differentially expressed genes cluster by sample group with similar expression patterns across all replicates, which they sort of do!

```{r heatmap-de fu, fig.height=6}
# Extract significant genes
sigGenes <- sig_genes_FU$gene

### Extract normalized expression for significant genes 
normalized_counts <- counts(dds, normalized=TRUE)
norm_sig <- normalized_counts[sigGenes,] 
meta <- metadata2$Dx_FUtime.x %>% data.frame()
rownames(meta) <- colnames(norm_sig)
### Set a color palette
heat_colors <- colorRampPalette(c("blue", "black", "yellow"))(100)
anno_colors <- list(condition=c(KO="#20A387FF",control="#453788FF"))

### Run pheatmap using the metadata data frame for the annotation (11 x 5)
pheatmap(norm_sig, 
    color = heat_colors, 
    cluster_rows = T, 
    show_rownames = F,
    annotation = meta, 
    annotation_colors = anno_colors,
    border_color = NA, 
    fontsize = 10, 
    scale = "row", 
    fontsize_row = 10, 
    height = 20)
```



## Heatmap of expression - Race

This plot shows only differentially expressed genes (padj < 0.1) on a per-sample basis. We have scaled the data by row and used the `ward.D2` method for clustering. We hope to see our differentially expressed genes cluster by sample group with similar expression patterns across all replicates, which they don't really.

```{r heatmap-de race, fig.height=6}
# Extract significant genes
sigGenes <- sig_genes_race$gene

### Extract normalized expression for significant genes 
normalized_counts <- counts(dds, normalized=TRUE)
norm_sig <- normalized_counts[sigGenes,] 
meta <- metadata2$Race.x %>% data.frame()
rownames(meta) <- colnames(norm_sig)
### Set a color palette
heat_colors <- colorRampPalette(c("blue", "black", "yellow"))(100)
anno_colors <- list(condition=c(KO="#20A387FF",control="#453788FF"))

### Run pheatmap using the metadata data frame for the annotation (11 x 5)
pheatmap(norm_sig, 
    color = heat_colors, 
    cluster_rows = T, 
    show_rownames = F,
    annotation = meta, 
    annotation_colors = anno_colors,
    border_color = NA, 
    fontsize = 10, 
    scale = "row", 
    fontsize_row = 10, 
    height = 20)
```


## Heatmap of expression - Surgery type 2 vs. 1

This plot shows only differentially expressed genes (padj < 0.1) on a per-sample basis. We have scaled the data by row and used the `ward.D2` method for clustering. We hope to see our differentially expressed genes cluster by sample group with similar expression patterns across all replicates, which they do not. 

```{r heatmap-de surgery, fig.height=6}
# Extract significant genes
sigGenes <- sig_genes_surgery$gene

### Extract normalized expression for significant genes 
normalized_counts <- counts(dds, normalized=TRUE)
norm_sig <- normalized_counts[sigGenes,] 
meta <- as.factor(metadata2$Surgery_Type.x) %>% data.frame()
rownames(meta) <- colnames(norm_sig)
### Set a color palette
heat_colors <- colorRampPalette(c("blue", "black", "yellow"))(100)
anno_colors <- list(condition=c(KO="#20A387FF",control="#453788FF"))

### Run pheatmap using the metadata data frame for the annotation (11 x 5)
pheatmap(norm_sig, 
    color = heat_colors, 
    cluster_rows = T, 
    show_rownames = F,
    annotation = meta, 
    annotation_colors = anno_colors,
    border_color = NA, 
    fontsize = 10, 
    scale = "row", 
    fontsize_row = 10, 
    height = 20)
```

### Top genes with Age

Here, we've plotted the abundance (DESeq2-normalized counts) of all differentially expressed genes (DEGs, ranked by adjusted p-value) in each sample. We plot age vs. abundance and add a trendline for the top 20 genes. Some of these look convincing but others do not.

```{r, fig.height=3, fig.width=3}

to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)
#Graph ones with a SIDS effect

rna = sig_genes_age$gene
names = sig_genes_age$external_gene_name

out <- vector("list", length = 20)

to_graph = data.frame(to_test[,rna])
to_graph$age <- metadata2$Age_DCIS_Dx.x


for (i in seq(1,20)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=age,y=temp))  + geom_point()+ ylab(paste0(names[[i]])) + geom_smooth() + theme(legend.position = "NONE")

  
}

out

```



### Top genes with Dx.FUtime

Here, we've plotted the abundance (DESeq2-normalized counts) of all differentially expressed genes (DEGs, ranked by adjusted p-value) in each sample. We plot Dx.FUtime vs. abundance and add a trendline for the top 20 genes. Some of these look convincing but others do not.

```{r, fig.height=3, fig.width=3}

to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)
#Graph ones with a SIDS effect

rna = sig_genes_FU$gene
names = sig_genes_FU$external_gene_name

out <- vector("list", length = 20)

to_graph = data.frame(to_test[,rna])
to_graph$Dx.FUtime <- metadata2$Dx_FUtime.x


for (i in seq(1,20)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=Dx.FUtime,y=temp))  + geom_point()+ ylab(paste0(names[[i]])) + geom_smooth() + theme(legend.position = "NONE")

  
}

out


```



### Top genes with Race

Here, we've plotted the abundance (DESeq2-normalized counts) of all differentially expressed genes (DEGs, ranked by adjusted p-value) in each sample. We plot Race vs. abundance. Some of these look convincing but others do not.

```{r, fig.height=3, fig.width=3}

to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)
#Graph ones with a SIDS effect

rna = sig_genes_race$gene
names = sig_genes_race$external_gene_name

out <- vector("list", length = 20)

to_graph = data.frame(to_test[,rna])
to_graph$race <- metadata2$Race.x


for (i in seq(1,20)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=race,y=temp)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[i]])) + geom_smooth() + theme(legend.position = "NONE")

  
}

out

```

### Top genes with Surgery

Here, we've plotted the abundance (DESeq2-normalized counts) of all differentially expressed genes (DEGs, ranked by adjusted p-value) in each sample. We plot surgery vs. abundance. Some of these look convincing but others do not.

```{r, fig.height=3, fig.width=3}

to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)
#Graph ones with a SIDS effect

rna = sig_genes_surgery$gene
names = sig_genes_surgery$external_gene_name

out <- vector("list", length = 20)

to_graph = data.frame(to_test[,rna])
to_graph$surgery <- metadata2$Surgery_Type.x


for (i in seq(1,20)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=surgery,y=temp)) + geom_boxplot() + geom_point(alpha=0.5)+  ylab(paste0(names[[i]])) + geom_smooth() + theme(legend.position = "NONE")

  
}

out

```





# R session

List and version of tools used for the QC report generation.

```{r}
sessionInfo()
```