---
title: "Differential Expression"
author: "Emma Berdan"
date: "`r Sys.Date()`"
output:
   html_document:
      code_folding: hide
      df_print: paged
      highlights: pygments
      number_sections: true
      self_contained: true
      theme: default
      toc: true
      toc_float:
         collapsed: true
         smooth_scroll: true
---

# Overview

- Principal Investigator: Aditi Hazra
- Experiment: Breast Cancer patient samples

```{r, cache = FALSE, message = FALSE, warning=FALSE}
library(tidyverse)
library(knitr)
library(DESeq2)
library(DEGreport)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
library(biomaRt)
library(org.Mm.eg.db)
library(topGO)
library(KEGGREST)
library(Category)
library(ape)
library(data.table)
library(fgsea)
library(clusterProfiler)
library(sva)
library(gprofiler2)
library(janitor)
library(ggvenn)
library(ggplot2)
ggplot2::theme_set(theme_light(base_size = 14))
opts_chunk[["set"]](
    cache = FALSE,
    dev = c("png", "pdf"),
    error = TRUE,
    highlight = TRUE,
    message = FALSE,
    prompt = FALSE,
    tidy = FALSE,
    warning = FALSE,
    fig.height = 4)

sanitize_datatable = function(df, ...) {
 # remove dashes which cause wrapping
 DT::datatable(df, ..., rownames=gsub("-", "_", rownames(df)),
                   colnames=gsub("-", "_", colnames(df)))
}
```

# Load and pre-process data

We load our data. We look at how many zeros we have per gene and see that for many genes it is very high. We set a cutoff that at most 25 samples can have zero counts.


## Get annotations
```{r get gene_annotations}
 # library(biomaRt)
#

# Get gene annotations based on reference data version used for alignment/quantification from BiomaRt

# Specify the Ensembl release ## dataset used by bcbio (check archives if needed)
#
# # List current versions of datasets
#  ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL")
#  datasets <- listDatasets(ensembl)
# #
# # # Identify the proper archive to use for corresponding Ensembl release
#  archives <- listEnsemblArchives()
# #
# # # # Use ensembl GRCm39
#   ensembl <- useMart(biomart = "ENSEMBL_MART_ENSEMBL",
#                     dataset = "mmusculus_gene_ensembl")
# # #
# # # ## Build biomaRt query
#  filters = listFilters(ensembl)
#  attributes = listAttributes(ensembl)
# # #
#   gene_annotations <- getBM(attributes=c('ensembl_gene_id', 'entrezgene_id',  'external_gene_name', 'description'),
#         filters = 'ensembl_gene_id',
#        values = rownames(counts),
#         mart = ensembl)
# #
# # #remove duplicated ensembl ids with different entrez gene IDs.
#  pruned_annotations = gene_annotations %>%
#    arrange(ensembl_gene_id, ensembl_gene_id) %>%
#    distinct(ensembl_gene_id, .keep_all = TRUE)
# #
#  write.csv(pruned_annotations, "/Users/emb016/Documents/HBC04572_Rosen/data/biomart.csv")
#
 pruned_annotations <- read.csv("/Users/emb016/Documents/HBC04676_Sicinski/annotation/biomart.csv") %>% .[,-1]
```

## Load Data

```{r  show-metadata}
se <- readRDS("/Users/emb016/Documents/HBC04292_Hazra/full_data/bcbio-se.rds")

metadata <- colData(se) %>% 
    as_tibble(rownames = NULL)

metrics <- metadata(se)$metrics %>% 
    left_join(metadata, by = c("sample" = "sample"))

meta_sm <- metadata %>%
  as.data.frame() %>%
  column_to_rownames("sample")


raw_counts <- assays(se)[["raw"]] 



coding_genes <- read.table("/Users/emb016/Documents/HBC04292_Hazra/lnc_and_protein_hg38.txt", header=FALSE)

raw_counts2 <- subset(raw_counts, rownames(raw_counts) %in% coding_genes$V2) 



## Remove WU104 and WU228

raw_counts3 <- raw_counts2[,-c(3 , 96)] %>% as_tibble()  %>% 
    as.matrix()

rownames(raw_counts3) <- rownames(raw_counts2)


meta <- data.frame(metadata[-c(3 , 96),])
met <- data.frame(metrics[-c(3 , 96),])

full2 <- meta %>% left_join(met, by="sample")
rownames(full2) <- full2$sample



## There are many zero heavy genes take a look at the number of zeros per gene
zero_dist <- rowSums(raw_counts3>0)
hist(zero_dist)

index <- which(zero_dist>116)
bad <- which(zero_dist<=116)
raw_counts4 <- raw_counts3[index,] 


vst <- vst(raw_counts4) 

genes_removed <- data.frame(rownames(raw_counts3[bad,]))
colnames(genes_removed) <- "gene"

genes_removed <- genes_removed %>% left_join(pruned_annotations, by=c("gene"="ensembl_gene_id"))

write.table(genes_removed, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/genes_removed_for_zeros.csv", sep=",", row.names = FALSE)
```


## Subset only cases

The number of zeros is a normal distribution and no factor seems to affect the distribution in a strong manner.

```{r}
meta_case <- subset(met, met$CaseControl==1)
counts_case <- raw_counts4[,colnames(raw_counts4) %in% meta_case$sample]

vst <- vst(counts_case) 

```




## Look at covariates

We only consider covariates that correlate with PC's explaining 3% or more of the variance for inclusion in the model. These are: Percent GC, Intronic Rate, Grade, Dx_FUtime, Age, and surgery type. 

**As intronic and exonic rate are usually correlated we will just use intronic rate. As age and menopause are correlated we will use age. **

```{r,fig.height=12, fig.width=8}

to_use <- meta_case[,c(1,3,5,7,8,9,12,18,24,25,26,27,30,31)]

rownames(to_use) <- to_use$sample

degCovariates(vst, metadata = to_use,minPC = 3)
```

## Scale continuous variables

Scaling is better for DESEQ2 and we scale total_reads, Dx_FUtime, and percent GC.

We will model inv_dx as categorical data.

```{r}



meta_case$scaled_FU <- scale(meta_case$Dx_FUtime)
meta_case$scaled_gc <- scale(meta_case$percent_gc)
meta_case$scaled_reads <- scale(meta_case$total_reads)


meta_case$Inv_Dx <- as.factor(meta_case$Inv_Dx)



```






# DESEQ -   gene expression ~ Total Reads + Percent GC + Dx_FUtime + Grade + Inv_dx.   - M1

**Note that we are modelling Grade as a continuous variable since grade 2 is intermediate to grade 1 and 3**

Differential gene expression analysis of count data was performed using the Bioconductor R package, DESeq2, which fits the count data to a negative binomial model.

Prior to identifying differentially expressed genes, the data was fit to the model and the dispersion was assessed to explore the fit of the count data to the model.

Using this model we will run DESEQ2. We often look at the dispersion plot to get a good idea of whether or not our data is a good fit for the model. Dispersion is a metric for variance which also takes into consideration mean expression. A dispersion value is estimated for each individual gene, then 'shrunken' to a more accurate value based on expected variation for the mean value. Finally, the shrunken dispersion value is used in the final GLM fit. From this plot we see the unshrunken (black) and shrunken (blue) dispersion values for every gene and that:

There is an expected inverse relationship between dispersion and mean expression. However, overall dispersions are very high. This is probably due to the high number of zeros. ***The data are an ok fit for the model.***


```{r}

## make sure metadata order matches count colnames

metadata2 <- data.frame(meta_case[match(meta_case$sample,colnames(counts_case)),])
rownames(metadata2) <- metadata2$sample


dds <- DESeqDataSetFromMatrix(counts_case, 
                              colData = metadata2, 
                              design = ~  scaled_FU + scaled_gc + scaled_reads +  Grade +  Inv_Dx)


dds <- DESeq(dds)

# Dispersion plot
plotDispEsts(dds)
```



## Extract results 

Below we summarize the results for each contrast of interest. Significant DE genes are found in files with sig_genes in the title. Full results (p-values, log2 fold changes, adjusted p-values and normalized counts) for all genes tested for each contrast are found in files with all_results in the title.

### Inv_dx

Using a Wald test, we tested for significant differences related to age using an adjusted p-value cut-off of 0.1. Since we ran this as a continuous variable the log2fold changes are per unit of the variable so can seem small. **Positive logFC means that expression is higher in inviduals where InvDx =1 and negative logFC means expression is higher in indviduals where InvDx=0.** 

There were 3 significant differentially expressed genes identified, of these genes 1 was up-regulated and 2 were down-regulated. 



```{r}

## get normalized counts
norm_c <- data.frame(counts(dds, normalized=TRUE))
norm_c$gene <- rownames(norm_c)


shrunken_res_Invdx <- data.frame(results(dds,
         name="Inv_Dx_1_vs_0"))


# Create a results tibble for treatment
res_tb_Invdx <- shrunken_res_Invdx %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  left_join(pruned_annotations, 
            by= c("gene" =  "ensembl_gene_id")) %>% 
  left_join(norm_c, 
            by= c("gene"))

res_tb_Invdx <- res_tb_Invdx[res_tb_Invdx$baseMean>0,]



# Subset to return genes with padj < 0.05

sig_genes_Invdx <- res_tb_Invdx %>%
        dplyr::filter(padj < 0.1) %>%
        arrange(padj)



#write out results
write.table(sig_genes_Invdx, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/sig_genes_Invdx_p0.1_casesonly.csv", sep=",", row.names = FALSE)
write.table(res_tb_Invdx, file="/Users/emb016/Documents/HBC04292_Hazra/full_data/Results/all_results_Invdx_casesonly.csv", sep=",", row.names = FALSE)

```


## Volcano plots

Here, we plot the log2 foldchange of the genes against the log10 adjusted p-value. The points highlighted in red are genes that have padj < 0.1 and a log2 foldchange > 0.1. Points in blue have a padj < 0.1 and a log2 foldchange < 0.1 and points in green have a padj > 0.1 and a log2 foldchange > 0.1. Grey points are non-significant. The dashed lines correspond to the cutoff values of log2 foldchance and padj that we have chosen.


```{r volcano-plots yumm m1, fig.align="center",fig.height=5, fig.width=6.5}
library("EnhancedVolcano")
EnhancedVolcano(res_tb_Invdx,
                lab= res_tb_Invdx$external_gene_name, pCutoff = 1.800562e-05
, selectLab = c(sig_genes_Invdx$external_gene_name[1:3]), FCcutoff = 0.1,
    x = 'log2FoldChange',
    y = 'pvalue', title="Inv.Dx", subtitle = "", drawConnectors = T)



```


## Heatmap of expression - Invdx

This plot shows only differentially expressed genes (padj < 0.1) on a per-sample basis. We have scaled the data by row and used the `ward.D2` method for clustering. We hope to see our differentially expressed genes cluster by sample group with similar expression patterns across all replicates, which they sort of do!

```{r heatmap-de age, fig.height=6}
# Extract significant genes
sigGenes <- sig_genes_Invdx$gene

### Extract normalized expression for significant genes 
normalized_counts <- counts(dds, normalized=TRUE)
norm_sig <- normalized_counts[sigGenes,] 
meta <- metadata2$Inv_Dx %>% data.frame()
rownames(meta) <- colnames(norm_sig)
### Set a color palette
heat_colors <- colorRampPalette(c("blue", "black", "yellow"))(100)
anno_colors <- list(condition=c(KO="#20A387FF",control="#453788FF"))

### Run pheatmap using the metadata data frame for the annotation (11 x 5)
pheatmap(norm_sig, 
    color = heat_colors, 
    cluster_rows = T, 
    show_rownames = F,
    annotation = meta, 
    annotation_colors = anno_colors,
    border_color = NA, 
    fontsize = 10, 
    scale = "row", 
    fontsize_row = 10, 
    height = 20)
```




### Top genes with InvDx

Here, we've plotted the abundance (DESeq2-normalized counts) of the top 3 differentially expressed genes (DEGs, ranked by adjusted p-value) in each sample. 

```{r, fig.height=3, fig.width=3}

to_test <- t(counts(dds, normalized=TRUE))

library(gridExtra)

rna = sig_genes_Invdx$gene
names = sig_genes_Invdx$external_gene_name

out <- vector("list", length = 3)

to_graph = data.frame(to_test[,rna])
to_graph$Inv_Dx <- metadata2$Inv_Dx

for (i in seq(1,3)) {
  
  to_graph$temp=to_graph[[i]]
  
  out[[i]] <- ggplot(to_graph,aes(x=Inv_Dx,y=temp)) + geom_boxplot() + geom_point() + ylab(paste0(names[[i]])) + theme(legend.position = "NONE") 

  
}

out

```



# R session

List and version of tools used for the QC report generation.

```{r}
sessionInfo()
```