---
title: "Quality Control"
author: "Emma Berdan"
date: "`r Sys.Date()`"
output:
   html_document:
      code_folding: hide
      df_print: paged
      highlights: pygments
      number_sections: true
      self_contained: true
      theme: default
      toc: true
      toc_float:
         collapsed: true
         smooth_scroll: true
---

# Overview

- Principal Investigator: Aditi Hazra
- Experiment: Breast Cancer patient samples

```{r, cache = FALSE, message = FALSE, warning=FALSE}
library(tidyverse)
library(knitr)
library(DESeq2)
library(DEGreport)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
ggplot2::theme_set(theme_light(base_size = 14))
opts_chunk[["set"]](
    cache = FALSE,
    dev = c("png", "pdf"),
    error = TRUE,
    highlight = TRUE,
    message = FALSE,
    prompt = FALSE,
    tidy = FALSE,
    warning = FALSE,
    fig.height = 4)
```

# Metadata

At the moment, we are only considering Case vs. Control as the main factor in the analysis.

```{r  show-metadata}
se <- readRDS("/Users/emb016/Documents/HBC04292_Hazra/full_data/bcbio-se.rds")
metadata <- colData(se) %>% 
    as_tibble(rownames = NULL)

metrics <- metadata(se)$metrics %>% 
    left_join(metadata, by = c("sample" = "sample")) %>% arrange(CaseControl)

order <- metrics$sample

meta_sm <- metadata %>%
  as.data.frame() %>%
  column_to_rownames("sample")

meta_sm[,1, drop=F] %>% kable()
```

## Readability

Note that for readability we will split the samples into two groups for graphing below. Samples wu101-wu198 and samples wu199-wu300

```{r}
metrics <- metadata(se)$metrics %>% 
    left_join(metadata, by = c("sample" = "sample")) 

metrics1 <- metrics[1:71,] %>% arrange(CaseControl)
metrics2 <- metrics[72:143,] %>% arrange(CaseControl)
order1 <- metrics1$sample
order2 <- metrics2$sample

```



# Read metrics {.tabset}

In the plots in the tabs below, we are looking for a few specific charateristics as described in each tab and consistency between samples. 

## Total reads

The total number of reads vary greatly across the samples, but most samples have over 20 million reads (blue line). We can see that WU228 has very few reads and may be an outlier.

```{r plot_total_reads 1, fig.height=10}
metrics1 %>%
    ggplot(aes(x = factor(sample, level = order1), 
               y = total_reads/1e6L, 
               fill = as.factor(CaseControl))) +
    geom_bar(stat = "identity") +
    coord_flip() +
    geom_hline(yintercept=20, color = "blue") +
    scale_y_continuous(name = "mln reads") +
    geom_text(aes(label = floor(total_reads/1e6L)), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") + 
    ggtitle("Total reads") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75)))
```


```{r plot_total_reads 2, fig.height=10}
metrics2 %>%
    ggplot(aes(x = factor(sample, level = order2), 
               y = total_reads/1e6L, 
               fill = as.factor(CaseControl))) +
    geom_bar(stat = "identity") +
  geom_hline(yintercept=20, color = "blue") +
    coord_flip() +
    scale_y_continuous(name = "mln reads") +
    geom_text(aes(label = floor(total_reads/1e6L)), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") + 
    ggtitle("Total reads") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75)))
```



## Mapping rate

The genomic mapping rate represents the percentage of reads mapping to the reference genome. We want to see consistent mapping rates between samples and over 70% mapping (blue line) but ideally at 90% reads mapping (red line).

```{r plot_mapping_rate 1, fig.height=10}
metrics1$mapped_reads_pct <- metrics1$mapped_reads/metrics1$total_reads
metrics1 %>%
    ggplot(aes(x = factor(sample, level = order1), 
               y = mapped_reads_pct, fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    coord_flip() +
    geom_text(aes(label = floor(mapped_reads_pct)), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") +
    ylim(0, 1.0) +
    ggtitle("Mapping rate") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.7))) +
  geom_hline(yintercept=0.7, color = "blue") + geom_hline(yintercept=0.9, color = "red")
```

```{r plot_mapping_rate 2, fig.height=10}
metrics2$mapped_reads_pct <- metrics2$mapped_reads/metrics2$total_reads
metrics2 %>%
    ggplot(aes(x = factor(sample, level = order2), 
               y = mapped_reads_pct, fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    coord_flip() +
    geom_text(aes(label = floor(mapped_reads_pct)), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") +
    ylim(0, 1.0) +
    ggtitle("Mapping rate") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.7))) +
  geom_hline(yintercept=0.7, color = "blue") + geom_hline(yintercept=0.9, color = "red")
```
## Number of genes detected

The number of genes represented in every sample is expected to be consistent and over 17K (blue line = 17,000). Many samples are under this but not by too much. Overall the data look good.

```{r plot_genes_detected 1, fig.height=10}
genes_detected <- colSums(assays(se)[["raw"]] > 0) %>% enframe()

sample_names <- metrics1[,c("sample"), drop=F]


genes_detected <- left_join(genes_detected, sample_names, by = c("name" = "sample"))

genes_detected <- genes_detected %>% group_by(name)

genes_detected <- summarise(genes_detected, 
                             n_genes = max(value))
                            
metrics1 <- metrics1 %>%
    left_join(genes_detected, by = c("sample" = "name"))
metrics1 %>% 
    ggplot(aes(x = factor(sample, level = order1),
               y = n_genes, fill = as.factor(CaseControl))) +
    geom_bar(stat = "identity") +
    coord_flip() +
    geom_text(aes(label = ""), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") + 
    ggtitle("Number of genes") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
  ylab("Number of genes") +
  xlab("") +
  geom_hline(yintercept=17000, color = "blue")
```


```{r plot_genes_detected 2, fig.height=10}
genes_detected <- colSums(assays(se)[["raw"]] > 0) %>% enframe()

sample_names <- metrics2[,c("sample"), drop=F]


genes_detected <- left_join(genes_detected, sample_names, by = c("name" = "sample"))

genes_detected <- genes_detected %>% group_by(name)

genes_detected <- summarise(genes_detected, 
                             n_genes = max(value))
                            
metrics2 <- metrics2 %>%
    left_join(genes_detected, by = c("sample" = "name"))
metrics2 %>% 
    ggplot(aes(x = factor(sample, level = order2),
               y = n_genes, fill = as.factor(CaseControl))) +
    geom_bar(stat = "identity") +
    coord_flip() +
    geom_text(aes(label = ""), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") + 
    ggtitle("Number of genes") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
  ylab("Number of genes") +
  xlab("") +
  geom_hline(yintercept=17000, color = "blue")
```


## Gene detection saturation

This plot shows how complex the samples are. Samples with more reads should have more genes. The pattern is there but weak, however, this is likely a result of the methodology used.

```{r plot_gene_saturation 1, fig.height=5, fig.width=5}
metrics1 %>% 
    ggplot(aes(x = log10(total_reads), 
               y = n_genes,
               color = as.factor(CaseControl))) +
        geom_point()+
        geom_text_repel(aes(label = sample))+
    ggtitle("Gene saturation") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
  ylab("Number of genes")
```


```{r plot_gene_saturation 2, fig.height=5, fig.width=5}
metrics2 %>% 
    ggplot(aes(x = log10(total_reads), 
               y = n_genes,
               color = as.factor(CaseControl))) +
        geom_point()+
        geom_text_repel(aes(label = sample))+
    ggtitle("Gene saturation") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
  ylab("Number of genes")
```

## Exonic mapping rate

Once again we are looking for consistency, and exonic mapping rates around 70% or 75% (blue and red lines, respectively). All of the samples look great with really high exonic rates. This means that majority of the mapped reads are exonic, which is reassuring.

```{r plot_exonic_mapping_rate 1, fig.height=10}
metrics1 %>%
    ggplot(aes(x = factor(sample, level = order1),
               y = exonic_rate * 100, 
               fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    geom_text(aes(label = floor(exonic_rate*100)), size = 2.5, hjust = 0, nudge_y = 0.5) +
    xlab("sample") + 
    ylab("Exonic rate %") + 
    ggtitle("Exonic mapping rate") + 
    coord_flip()  +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
  xlab("") +
  geom_hline(yintercept=70, color = "blue") +
  geom_hline(yintercept=75, color = "red")
```

```{r plot_exonic_mapping_rate 2, fig.height=10}
metrics2 %>%
    ggplot(aes(x = factor(sample, level = order2),
               y = exonic_rate * 100, 
               fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    geom_text(aes(label = floor(exonic_rate*100)), size = 2.5, hjust = 0, nudge_y = 0.5) +
    xlab("sample") + 
    ylab("Exonic rate %") + 
    ggtitle("Exonic mapping rate") + 
    coord_flip()  +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
  xlab("") +
  geom_hline(yintercept=70, color = "blue") +
  geom_hline(yintercept=75, color = "red")
```


## Intronic mapping rate

Here, we expect a low intronic mapping rate (â‰¤ 15% - 20%, red and blue lines). It seems that sometimes reads extended past the exons captured by the baits which is where these numbers are coming from. WU104 and WU279 do not look great here so we will keep an eye on it. 

```{r plot_intronic_mapping_rate 1, fig.height=10}
metrics1 %>%
    ggplot(aes(x = factor(sample, level = order1),
               y = intronic_rate * 100, 
               fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    geom_text(aes(label = floor(intronic_rate*100)), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") + 
    ylab("Intronic rate %") +
    ggtitle("Intronic mapping rate") + 
    coord_flip()  +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
    geom_hline(yintercept=20, color = "blue") +
    geom_hline(yintercept=15, color = "red")
```


```{r plot_intronic_mapping_rate 2, fig.height=10}
metrics2 %>%
    ggplot(aes(x = factor(sample, level = order2),
               y = intronic_rate * 100, 
               fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    geom_text(aes(label = floor(intronic_rate*100)), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") + 
    ylab("Intronic rate %") +
    ggtitle("Intronic mapping rate") + 
    coord_flip()  +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
    geom_hline(yintercept=20, color = "blue") +
    geom_hline(yintercept=15, color = "red")
```


## rRNA mapping rate

Samples should have a ribosomal RNA (rRNA) "contamination" rate below 10% and all of the samples look great!

```{r plot_rrna_mapping_rate 1, fig.height=10}
# for some bad samples it could be > 50%
rrna_ylim <- max(round(metrics1$r_rna_rate*100, 2)) + 10
metrics1 %>%
    ggplot(aes(x = factor(sample, level = order1),
               y = r_rna_rate * 100, 
               fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    geom_text(aes(label = round(r_rna_rate*100,2)), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") + 
    ylab("rRNA rate, %")+
    ylim(0, rrna_ylim) + 
    ggtitle("rRNA mapping rate") +
    coord_flip() +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75)))
```

```{r plot_rrna_mapping_rate 2, fig.height=10}
# for some bad samples it could be > 50%
rrna_ylim <- max(round(metrics2$r_rna_rate*100, 2)) + 10
metrics2 %>%
    ggplot(aes(x = factor(sample, level = order2),
               y = r_rna_rate * 100, 
               fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    geom_text(aes(label = round(r_rna_rate*100,2)), size = 2.5, hjust = 0, nudge_y = 0.5)+
    xlab("sample") + 
    ylab("rRNA rate, %")+
    ylim(0, rrna_ylim) + 
    ggtitle("rRNA mapping rate") +
    coord_flip() +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75)))
```



## 5'->3' bias

There should be little bias, i.e. the values should be close to 1. This is true in all samples here, they are all above 1, but not by much. 

```{r plot_53_bias 1, fig.height=10}
metrics1 %>%
    ggplot(aes(x = factor(sample, level = order1),
               y = x5_3_bias, 
               fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    geom_text(aes(label = x5_3_bias), size = 2.5, hjust = 0, nudge_y = 0.01) +
    xlab("sample") + 
    ggtitle("5'-3' bias") + 
    coord_flip() +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
    geom_hline(yintercept=1, color = "blue")
```


```{r plot_53_bias 2, fig.height=10}
metrics2 %>%
    ggplot(aes(x = factor(sample, level = order2),
               y = x5_3_bias, 
               fill = as.factor(CaseControl))) +
        geom_bar(stat = "identity") +
    geom_text(aes(label = x5_3_bias), size = 2.5, hjust = 0, nudge_y = 0.01) +
    xlab("sample") + 
    ggtitle("5'-3' bias") + 
    coord_flip() +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75))) +
    geom_hline(yintercept=1, color = "blue")
```


## Counts per gene - all genes

We expect consistency in the box plots here between the samples, i.e. the distribution of counts across the genes is similar. This plot looks good.

```{r plot_counts_per_gene 1, fig.height=10}
metrics_small1 <- metrics1 %>% dplyr::select(sample, CaseControl)

counts <- assays(se)[["raw"]] %>% 
    as_tibble() %>% 
    filter(rowSums(.)!=0) %>% 
    gather(name, counts)

counts <- left_join(counts, metrics_small1, by = c("name" = "sample"))

ggplot(counts, aes(factor(name, level = order1), log2(counts+1), fill = as.factor(CaseControl))) +
    geom_boxplot() + 
    coord_flip() +
    ggtitle("Counts per gene, all non-zero genes") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75)))
```

```{r plot_counts_per_gene 2, fig.height=10}
metrics_small2 <- metrics2 %>% dplyr::select(sample, CaseControl)

counts <- assays(se)[["raw"]] %>% 
    as_tibble() %>% 
    filter(rowSums(.)!=0) %>% 
    gather(name, counts)

counts <- left_join(counts, metrics_small2, by = c("name" = "sample"))

ggplot(counts, aes(factor(name, level = order2), log2(counts+1), fill = as.factor(CaseControl))) +
    geom_boxplot() + 
    coord_flip() +
    ggtitle("Counts per gene, all non-zero genes") +
    theme(plot.title=element_text(hjust=0.5), 
          axis.text.y = element_text(size = rel(0.75)))
```

# Sample similarity analysis 

In this section, we look at how well the different groups in the dataset cluster with each other. Samples from the same group should ideally be clustering together. We use 2 different methods - the first one is Principal Component Analysis (PCA) and the second one is hierarchical clustering with a correlation heatmap. **Here we are only using data from protein coding genes**

## Principal component analysis (PCA)


```{r pca}

coding_genes <- read.table("/Users/emb016/Documents/HBC04292_Hazra/genes_pc.txt", header=FALSE)

raw_counts <- assays(se)[["raw"]] 

raw_counts2 <- subset(raw_counts, rownames(raw_counts) %in% coding_genes$V2) %>% 
    as_tibble() %>% 
    filter(rowSums(.)!=0) %>% 
    as.matrix()

vst <- vst(raw_counts2) 

meta <- data.frame(colData(se))
batch <- read.csv("/Users/emb016/Documents/HBC04292_Hazra/full_data/batch_info.csv",header = TRUE)

full_meta <- meta %>% left_join(batch, by="sample")

## get rid of weird dup line
full_meta <- full_meta[-142,]
rownames(full_meta) <- full_meta$sample
colnames(full_meta)[30] <- "batch"  
  
pca <- degPCA(vst, full_meta, condition = "CaseControl", name = "sample", data = T)[["plot"]]

pca_labels <- pca[["labels"]]
pca_data <- pca[["data"]] %>% as_tibble()
pca_data %>% 
    ggplot(aes(x = PC1, y = PC2, color = as.factor(CaseControl))) + 
    geom_point() +  
    xlab(pca_labels$x) + 
    ylab(pca_labels$y) +
    ggtitle(paste0("All samples", "\nPCA using ", nrow(vst), " genes")) +
    theme(plot.title=element_text(hjust=0.5))


pca_data %>% 
    ggplot(aes(x = PC1, y = PC2, color = as.factor(batch))) + 
    geom_point() +  
    xlab(pca_labels$x) + 
    ylab(pca_labels$y) +
    ggtitle(paste0("All samples", "\nPCA using ", nrow(vst), " genes")) +
    theme(plot.title=element_text(hjust=0.5))

```



## Hierarchical clustering

In this case the heatmap represents correlation, and the dendograms along the x- and y- axes represent how similar/dissimilar the replicates for each samplegroup are to each other. 


We can see that a small subset of samples cluster away from everything else. These are wu104, wu222 wu118, wu127. We will remove these and redo the PCA and hierarchical clustering.
```{r clustering}

vst_cor <- cor(vst)


### Extract the vst matrix from the object
colnames(vst_cor) <- metadata$sample
rownames(vst_cor) <- metadata$sample

pheatmap(vst_cor, annotation = meta_sm[,5, drop=F], show_rownames = T, show_colnames = T)



```

# Sample similarity no wu104, wu222, wu118, wu127


## Principal component analysis (PCA)



```{r pca 2}



raw_counts3 <- raw_counts2[,-c(3 , 91  , 9 ,18)]
vst <- vst(raw_counts3) 

meta <- data.frame(full_meta[-c(3 , 91  , 9 ,18),])
met <- data.frame(metrics[-c(3 , 91  , 9 ,18),])

full2 <- meta %>% left_join(met, by="sample")
rownames(full2) <- full2$sample

pca <- degPCA(vst, full2, condition = "CaseControl", name = "sample", data = T)[["plot"]]

pca_labels <- pca[["labels"]]
pca_data <- pca[["data"]] %>% as_tibble()
pca_data %>% 
    ggplot(aes(x = PC1, y = PC2, color = as.factor(CaseControl.x))) + 
    geom_point() +  
    xlab(pca_labels$x) + 
    ylab(pca_labels$y) +
    ggtitle(paste0("All samples", "\nPCA using ", nrow(vst), " genes")) +
    theme(plot.title=element_text(hjust=0.5))


pca_data %>% 
    ggplot(aes(x = PC1, y = PC2, color = exonic_rate)) + 
    geom_point() +  
    xlab(pca_labels$x) + 
    ylab(pca_labels$y) +
    ggtitle(paste0("All samples", "\nPCA using ", nrow(vst), " genes")) +
    theme(plot.title=element_text(hjust=0.5))


pca_data %>% 
    ggplot(aes(x = PC1, y = PC2, color = intronic_rate)) + 
    geom_point() +  
    xlab(pca_labels$x) + 
    ylab(pca_labels$y) +
    ggtitle(paste0("All samples", "\nPCA using ", nrow(vst), " genes")) +
    theme(plot.title=element_text(hjust=0.5))

pca_data %>% 
    ggplot(aes(x = PC1, y = PC2, color = percent_gc)) + 
    geom_point() +  
    xlab(pca_labels$x) + 
    ylab(pca_labels$y) +
    ggtitle(paste0("All samples", "\nPCA using ", nrow(vst), " genes")) +
    theme(plot.title=element_text(hjust=0.5))



pca_data %>% 
    ggplot(aes(x = PC1, y = PC2, color = as.factor(batch.x.x))) + 
    geom_point() +  
    xlab(pca_labels$x) + 
    ylab(pca_labels$y) +
    ggtitle(paste0("All samples", "\nPCA using ", nrow(vst), " genes")) +
    theme(plot.title=element_text(hjust=0.5))
```

## Hierarchical clustering

We see clustering but cannot explain it yet.
```{r clustering 2}

vst_cor <- cor(vst)


### Extract the vst matrix from the object
colnames(vst_cor) <- meta$sample
rownames(vst_cor) <- meta$sample

pheatmap(vst_cor, annotation = meta_sm[,5, drop=F], show_rownames = T, show_colnames = T)

pheatmap(vst_cor, annotation = meta_sm[,9, drop=F], show_rownames = T, show_colnames = T)

pheatmap(vst_cor, annotation = meta_sm[,21, drop=F], show_rownames = T, show_colnames = T)

pheatmap(vst_cor, annotation = meta_sm[,19, drop=F], show_rownames = T, show_colnames = T)


pheatmap(vst_cor, annotation = meta_sm[,12, drop=F], show_rownames = T, show_colnames = T)

pheatmap(vst_cor, annotation = meta_sm[,13, drop=F], show_rownames = T, show_colnames = T)

pheatmap(vst_cor, annotation = meta_sm[,14, drop=F], show_rownames = T, show_colnames = T)



```


### Covariate correlation with PCs

Multiple factors related to the experimental design or quality of sequencing may influence the outcomes of a given RNA-seq experiment. To further determine whether any confounding covariate risks affecting the results of our differential expression analyses, it is useful to assess the correlation between covariates and principal component (PC) values.

Here, we are using `DEGreport::degCovariates()` to explore potential correlations between variables provided in the metadata and all PCs that account for at least 5% of the variability in the data. If applicable, significant correlations (FDR < 0.1) are circled . This diagnostic plot helps us determine which variables to keep an eye on in subsequent sample similarity analyses. 

**Note that this function won't plot covariates with NA values. So first we use all our our data to plot what we can then we subset the data to remove NAs (removing 10 samples) and plot again.**

```{r,cache = FALSE, message = FALSE, fig.height=4, error=FALSE, warning=FALSE, fig.height=10, fig.width=8}


full3 <- full2[,c(1,2,5,7,8,9,12,13,14,15,19,21,29,30,31,34,35,36,37)]

rownames(full3) <- full3$sample


degCovariates(vst, metadata = full3, minPC=1 )
```



```{r,cache = FALSE, message = FALSE, fig.height=4, error=FALSE, warning=FALSE, fig.height=10, fig.width=8}


full3 <- full2[,c(1,2,5,7,8,9,12,13,14,15,19,21,29,30,31,34,35,36,37)]

rownames(full3) <- full3$sample
full3 <- na.omit(full3)
idx <-  colnames(vst) %in% rownames(full3) 
vst2 <- vst[,idx]

degCovariates(vst2, metadata = full3, minPC=1 )
```

# R session

List and version of tools used for the QC report generation.

```{r}
sessionInfo()
```